setup.py

```diff
@@ -57,15 +57,45 @@ def run_tests(self):
 
     cmdclass['build_docs'] = BuildDoc
 
+__version__ = None # Explicitly set version to quieten static code checkers.
 exec(open('moviepy/version.py').read()) # loads __version__
 
 # Define the requirements for specific execution needs.
-requires = ['decorator==4.0.11', 'imageio==2.1.2', 'tqdm==4.11.2', 'numpy']
-optional_reqs = ['scikit-image==0.13.0', 'scipy==0.19.0', 'matplotlib==2.0.0']
-documentation_reqs = ['pygame==1.9.3', 'numpydoc>=0.6.0',
-    'sphinx_rtd_theme>=0.1.10b0', 'Sphinx>=1.5.2'] + optional_reqs
-test_reqs = ['pytest>=2.8.0', 'nose', 'sklearn', 'pytest-cov', 'coveralls'] \
-    + optional_reqs
+requires = [
+    'decorator>=4.0.2,<5.0',
+    'imageio>=2.1.2,<3.0',
+    'tqdm>=4.11.2,<5.0',
+    'numpy',
+    ]
+
+optional_reqs = [
+        "opencv-python>=3.0,<4.0; python_version!='2.7'",
+        "scikit-image>=0.13.0,<1.0; python_version>='3.4'",
+        "scikit-learn; python_version>='3.4'",
```

Comment:
Haven't looked at all of the changes here. So could be missing something. Is this suppose to be a test requirement still? Did it become an optional requirement as well?